command/helpers.go

```diff
@@ -606,6 +620,36 @@ func extractVarFlags(slice []string) map[string]string {
 	return m
 }
 
+// extractJobSpecEnvVars is used to extract Nomad specific HCL variables from
+// the OS environment. The input envVars parameter is expected to be generated
+// from the os.Environment function call. The result is never nil for
+// convenience.
+func extractJobSpecEnvVars(envVars []string) map[string]string {
+
+	m := make(map[string]string)
+
+	for _, raw := range envVars {
+		if !strings.HasPrefix(raw, jobspec2.VarEnvPrefix) {
+			continue
+		}
+
+		// Trim the prefix, so we just have the raw key=value variable
+		// remaining.
+		raw = raw[len(jobspec2.VarEnvPrefix):]
```

Comment:
`NOMAD_VAR_=blah` makes this `raw = "=blah"` so produces `m[""] = "blah"` below -- do we care to handle that edge case?

the non-env version looks like this

```
$ nomad run -var '=nah' vars.nomad.hcl
Error getting job struct: Error parsing job file from vars.nomad.hcl:
<nil>: Undefined -var variable; A "" variable was passed in the command line but was not found in known variables. To declare variable "", place this block in your job file
<nil>: Unset variable "one"; A used variable must be set or have a default value; see https://www.nomadproject.io/docs/job-specification/hcl2/variables for details.
```

which arguably is a kinda silly error since "" is not a valid variable.  maybe a silent no-op, as happens here, is preferable?